lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/mic.rb

```diff
@@ -0,0 +1,196 @@
+              BinData::Int32be.new(0x666d7420),      # SubChunk1ID: "fmt "
+              BinData::Int32le.new(16),              # SubChunk1Size
+              BinData::Int16le.new(1),               # AudioFormat
+              BinData::Int16le.new(num_channels),    # NumChannels
+              BinData::Int32le.new(sample_rate_hz),  # SampleRate
+              BinData::Int32le.new(byte_rate),       # ByteRate
+              BinData::Int16le.new(block_align),     # BlockAlign
+              BinData::Int16le.new(bits_per_sample), # BitsPerSample
+              BinData::Int32be.new(0x64617461),      # SubChunk2ID: "data"
+              BinData::Int32le.new(data_size)        # SubChunk2Size
+            ]
+          end
+
+          def cmd_mic_start(*args)
+            get_data = lambda do |channel, file|
+              data = channel.read(65536)
+              if data
+                ::File.open(file, 'a') do |f|
+                  f.write(data)
+                end
+                return data.length
+              end
+              return 0
+            end
+            device_id = 1
+            duration = 1800
+            saved_audio_path = Rex::Text.rand_text_alpha(8) + ".wav"
+
+            mic_start_opts = Rex::Parser::Arguments.new(
+              "-h" => [ false, "Help Banner" ],
+              "-d" => [ true, "The stream duration in seconds (Default: 1800)" ], # 30 min
+              "-m" => [ true, "Microphone device index to record from (1: system default)" ],
+              "-s" => [ true, "The saved audio file path (Default: '#{saved_audio_path}')" ]
+            )
+
+            mic_start_opts.parse(args) do |opt, _idx, val|
+              case opt
+                when "-h"
+                  print_line("Usage: mic_start [options]\n")
+                  print_line("Streams and records audio from the target microphone.")
+                  print_line(mic_start_opts.usage)
+                  return
+                when "-d"
+                  duration = val.to_i
+                when "-m"
+                  device_id = val.to_i
+                when "-s"
+                  saved_audio_path = val
+              end
+            end
+
+            mic_list = client.mic.mic_list
+            if mic_list.length == 0
+              print_error("Target does not have a mic")
+              return
+            end
+            if device_id < 1 || device_id > mic_list.length
+              print_error("Target does not have a mic with an id of #{device_id}")
+              return
+            end
+
+            print_status("Saving to audio file: #{saved_audio_path}")
+            total_data_len = 0
+            begin
+              channel = client.mic.mic_start(device_id)
+              mic_started = true
+              print_status("Streaming started...")
+              ::File.open(saved_audio_path, 'wb') do |outfile|
+                audio_file_wave_header(11025, 1, 16, 2000000000).each { |e| e.write(outfile) }
```

Comment:
I greatly prefer keyword arguments wherever possible. I'm also a big fan of using `_` as a delimiter in large number literals. I'd do it like this:

```ruby
audio_file_wave_header(sample_rate_hz: 11025, num_channels: 1, bits_per_sample: 16, data_size: 2_000_000_000).each do |e|
  e.write(outfile)
end